File: domain/src/main/java/co/crisi/wardrobe/wizard/api/domain/data/Item.java
package co.crisi.wardrobe.wizard.api.domain.data;

import co.crisi.wardrobe.wizard.api.domain.IItem;
import co.crisi.wardrobe.wizard.api.domain.validator.decorator.NonNullValidatorDecorator;
import co.crisi.wardrobe.wizard.api.domain.validator.decorator.Validator;

public record Item(Long id, String name, String description, String imageUrl,
                   String color, String size, String brand, String material,
                   ItemCategory category, ItemType type) implements IItem {


    public Item(Long id, String name, String description, String imageUrl,
                String color, String size, String brand, String material,
                ItemCategory category, ItemType type) {

        var validator = new Validator();
        var nonNullValidator = new NonNullValidatorDecorator(validator);

        this.id = nonNullValidator.validate(id, "item id");
        this.name = nonNullValidator.validate(name, "item name");
        this.description = nonNullValidator.validate(description, "item description");
        this.imageUrl = nonNullValidator.validate(imageUrl, "item image url");
        this.color = nonNullValidator.validate(color, "item color");
        this.size = nonNullValidator.validate(size, "item size");
        this.brand = nonNullValidator.validate(brand, "item brand");
        this.material = nonNullValidator.validate(material, "item material");
        this.category = nonNullValidator.validate(category, "item category");
        this.type = nonNullValidator.validate(type, "item type");

    }

    @Override
    public String getName() {
        return name;
    }

    @Override
    public String getDescription() {
        return description;
    }

    @Override
    public String getImageUrl() {
        return imageUrl;
    }

    @Override
    public String getColor() {
        return color;
    }

    @Override
    public String getSize() {
        return size;
    }

    @Override
    public String getBrand() {
        return brand;
    }

    @Override
    public String getMaterial() {
        return material;
    }

    @Override
    public ItemCategory getCategory() {
        return category;
    }

    @Override
    public ItemType getType() {
        return type;
    }

    @Override
    public Long getId() {
        return id;
    }
}

File: src/domain/entities/item.py
class Item:
    
    def __init__(self, id:str, name:str, category: str, color:str, style: str, brand:str, season: str, image_url: str):
        self.id = id
        self.name = name
        self.category = category
        self.color = color
        self.style = style
        self.brand = brand
        self.season = season
        self.image_url = image_url
        
    def __str__(self):
        return f"Item(id={self.id}, name={self.name}, category={self.category}, color={self.color}, style={self.style}, brand={self.brand}, season={self.season}, image_url={self.image_url})"

    def to_dict(self):
        return {
            "id": str(self.id),
            "name": self.name,
            "category": self.category,
            "color": self.color,
            "style": self.style,
            "brand": self.brand,
            "season": self.season,
            "image_url": self.image_url
        }